drift/inventory_service_interface.py

```diff
@@ -49,18 +49,19 @@ def _fetch_url(url, auth_header, logger):
     return response.json()
 
 
-def _fetch_paginated_url(url, auth_header, logger):
+def _fetch_data(url, auth_header, system_ids, logger):
     """
-    helper to collate a paginated response
+    fetch system IDs in batches of 20 for given RESTful URL
     """
+    BATCH_SIZE = 20
```

Comment:
Is this the most optimal value?

Reply:
I believe 53 was as many as I could get before I hit the URL limit. I chose 20 for human and infra purposes moreso than performance. I did not benchmark various batch sizes, but I was able to get 3K comparisons in 90 sec with batch size of 20.

Reply:
I added a comment to the code to capture this so it doesn't get lost later

Reply:
Are you able to run the 3k test fairly easily? Would you be up for testing 50 and 100 for batch sizes?

Reply:
test results for 2K systems:

20 = 2m10s
40 = 1m40s
50 = 1m30s

I changed this to 40 to leave some overhead for URL length limits.

Reply:
Glad to see there was improvements for larger batch sizes. Seems to be trending towards smaller lengths. I am good with sticking to 40 for now.